rp2040-hal/src/float/cmp.rs

```diff
@@ -0,0 +1,198 @@
+use super::Float;
+use crate::rom_data;
+
+trait ROMCmp {
+    fn rom_cmp(self, b: Self) -> i32;
+}
+
+impl ROMCmp for f32 {
+    fn rom_cmp(self, b: Self) -> i32 {
+        rom_data::float_funcs::fcmp(self, b)
+    }
+}
+
+impl ROMCmp for f64 {
+    fn rom_cmp(self, b: Self) -> i32 {
+        rom_data::double_funcs::dcmp(self, b)
+    }
+}
+
+fn le_abi<F: Float + ROMCmp>(a: F, b: F) -> i32 {
+    if a.is_nan() || b.is_nan() {
+        1
+    } else {
+        a.rom_cmp(b)
+    }
+}
+
+fn ge_abi<F: Float + ROMCmp>(a: F, b: F) -> i32 {
+    if a.is_nan() || b.is_nan() {
+        -1
+    } else {
+        a.rom_cmp(b)
+    }
+}
+
+intrinsics! {
+    #[slower_than_default]
+    #[bootrom_v2]
+    #[alias = __eqsf2, __ltsf2, __nesf2]
```

Comment:
I don't understand those aliases. Aren't those different operations (equal, less than, not equal and less-or-equal?)

Reply:
Yes, but the [definition of the functions](https://gcc.gnu.org/onlinedocs/gccint/Soft-float-library-routines.html) is things like "zero if strictly equal" or "negative if less" so the allow any other value if the condition isn't met.  [compiler-builtins does the same thing](https://github.com/rust-lang/compiler-builtins/blob/9c4118f197fe8cb2b579555a497f5df6b3b3afc3/src/float/cmp.rs#L123), which is actually where I got the logic from.

Reply:
Ok, fine, I think now I understand: The underlying function (bootrom function `_fcmp`, softfloat library call `__cmpsf2`) distinguishes three cases, `<`, `==`, `>`. The library also provides several alternatives which don't guarantee to distinguish all three cases, but just separate subsets. I guess they are separate in case the hardware allows for faster implementations if you are only interested in such a subset.
Thanks for those pointers!